OpenRA.Mods.Common/Traits/Conditions/ConditionManager.cs

```diff
@@ -228,14 +196,10 @@ public bool AcceptsExternalCondition(Actor self, string condition, bool timed =
 				return false;
 
 			// A timed condition can always replace an existing timed condition (resetting its duration)
-			if (timed && timers.ContainsKey(condition))
-				return true;
-
-			string[] sc;
-			if (stackedConditions.TryGetValue(condition, out sc))
-				return stackedTokens[condition].Count < sc.Length;
+			if (timed)
+				return timers.ContainsKey(condition);
 
-			return conditionCache[condition] == 0;
+			return true;
```

Comment:
This method can now simplify down to `return externalConditions.Contains(condition);`